packages/nuxt/src/pages/runtime/router.options.ts

```diff
@@ -68,23 +60,23 @@ function _getHashElementScrollMarginTop (selector: string): number {
 
 function _calculatePosition (
   to: RouteLocationNormalized,
-  scrollBehaviorType: ScrollBehavior,
-  position?: ScrollPosition,
+  from: RouteLocationNormalized,
+  savedPosition: ScrollPosition | null,
 ): ScrollPosition {
-  // Handle saved position for backward/forward navigation
-  if (position) {
-    return position
+  if (!savedPosition && isChangingPage(to, from) && !to.hash) {
+    return { left: 0, top: 0 }
+  }
+
+  if (savedPosition) {
+    return savedPosition
   }
 
-  // Scroll to the element specified in the URL hash, if present
   if (to.hash) {
     return {
       el: to.hash,
       top: _getHashElementScrollMarginTop(to.hash),
-      behavior: scrollBehaviorType,
```

Comment:
I think the 'instant' behaviour is necessary when user has specified smooth scroll behaviour, to avoid smooth scroll to a saved position. in that case (in initial load), it should be instant.